src/cpp/include/openvino/genai/vision_encoder.hpp

```diff
@@ -41,11 +41,16 @@ struct EncodedImage {
 /// ov::InferRequest and configured by ProcessorConfig.
 class OPENVINO_GENAI_EXPORTS VisionEncoder {
 public:
+    /// @brief A string denoting model type.
+    std::string model_type;
     /// @brief A model for image encoding.
     ov::InferRequest m_encoder;
     /// @brief A config to follow.
     ProcessorConfig m_processor_config;
 
+    // LLaVa specific members
+    ov::InferRequest m_vision_embeddings;
```

Comment:
should we define base class for VisionEncoder ? and create inherited classes for MiniCPM and LLava specifically 

In this case, we don't need to put all the fields for different models into common class. Similarly, processor config can be model specific class.